File: backend/server/responses.py
import typing as t
from datetime import datetime

from pydantic import BaseModel


class DefaultResponse(BaseModel):
    message: str


class UserResponse(BaseModel):
    name: str
    username: str
    is_admin: bool


class PostResponse(BaseModel):
    id: int
    title: str
    content: str
    creator: UserResponse


class CommentResponse(BaseModel):
    id: int
    content: str
    creator: UserResponse

File: frontend/types/responses.ts
interface ErrorData {
  loc: string[];
  msg: string;
  type: string;
}

interface ErrorResponse {
  detail: string | ErrorData[];
}

interface DefaultResponse {
  message: string;
}

interface UserResponse {
  name: string;
  username: string;
  is_admin: boolean;
}

interface PostResponse {
  id: number;
  title: string;
  content: string;
  creator: UserResponse;
}

interface CommentResponse {
  id: number;
  content: string;
  creator: UserResponse;
}

export type {
  ErrorData,
  ErrorResponse,
  DefaultResponse,
  UserResponse,
  PostResponse,
  CommentResponse,
};
